### DISCUSSION TITLE
getCss({ component }) returns the full CSS string for the whole editor. From the documentation it appeared it should return the CSS for just the component passed into the function..

### DISCUSSION THREAD

**Discussion by SamuelVirgo** [GrapesJS/grapesjs#4346] [Q&A]
Hi,

I think I have misunderstood the documentation:

https://grapesjs.com/docs/api/editor.html#getcss

Here, it seems we should be able to use this function to get CSS for the selected component like this:

```
var opts = {};
opts.component = editor.getSelected();
const css = editor.getCss(opts);
```

Due to the fact that opts.component returns "(...) the CSS of a specific Component".

However, when I test this it returns the css for the whole editor still. Can someone help me where I am going wrong?

**Comment by artf** [comment]
I see it working properly, please create a reproducible demo of your issue.
